src/benchmark_environments/util.py

```diff
@@ -17,3 +22,183 @@ def step(self, action):
             info["terminal_observation"] = obs
             obs = self.env.reset()
         return obs, rew, False, info
+
+
+class AbsorbAfterDoneWrapper(gym.Wrapper):
+    """Goes to absorb state after done=True. Always returns done=False.
+
+    A convenient way to add absorbing states to environments like MountainCar.
+    """
+
+    def __init__(
+        self,
+        env: gym.Env,
+        absorb_reward: float = 0.0,
+        absorb_obs: Optional[np.ndarray] = None,
+    ):
+        """Initialize AbsorbAfterDoneWrapper.
+
+        Args:
+          env: The wrapped Env.
+          absorb_reward: The reward returned at the absorb state.
+          absorb_obs: The observation returned at the absorb state. If None, then
+            repeat the final observation before absorb.
+        """
+        super().__init__(env)
+        self.absorb_reward = absorb_reward
+        self.absorb_obs_default = absorb_obs
+        self.absorb_obs_this_episode = None
+        self.at_absorb_state = None
+
+    def reset(self, *args, **kwargs):
+        """Reset the environment."""
+        self.at_absorb_state = False
+        self.absorb_obs_this_episode = None
+        return self.env.reset(*args, **kwargs)
+
+    def step(self, action):
+        """Advance the environment by one step.
+
+        This wrapped `step()` always returns done=False.
+
+        After the first done is returned by the underlying Env, we enter an artificial
+        absorb state. In this artificial absorb state, we stop calling
+        `self.env.step(action)` (i.e. the `action` argument is entirely ignored) and
+        we return fixed values for obs, rew, done, and info.
+        """
+        if not self.at_absorb_state:
+            inner_obs, inner_rew, done, inner_info = self.env.step(action)
+            if done:
+                # Initialize the artificial absorb state, which we will repeatedly use
+                # starting on the next call to `step()`.
+                self.at_absorb_state = True
+
+                if self.absorb_obs_default is None:
+                    self.absorb_obs_this_episode = inner_obs
+                else:
+                    self.absorb_obs_this_episode = self.absorb_obs_default
+            obs, rew, info = inner_obs, inner_rew, inner_info
+        else:
+            assert self.absorb_obs_this_episode is not None
+            assert self.absorb_reward is not None
+            obs = self.absorb_obs_this_episode
+            rew = self.absorb_reward
+            info = {}  # XXX: Also allow custom info for absorb state.
```

Comment:
No info in absorbing state doesn't seem like a big deal.

If you want to leave a comment please make it `TODO(shwang): ...`

Generally todo's should always have some source of info for context. Ideally a GitHub issue, but a username better than nothing.

Reply:
Removed the todo comment and added to docstring that info dict is always empty.